### PR TITLE
[DomCrawler] Improve the sprintf() call in selectButton()

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  |no 
| BC breaks?    | no     
| Deprecations? |no 
| Tests pass?   | yes    
| Fixed tickets | n/a   
| License       | MIT
| Doc PR        | n/a

Code cleanup and minor performance improvement. Similar to #26485.

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -738,12 +738,14 @@ public function selectImage($value)
...
     public function selectButton($value)
     {
-        $translate = 'translate(@type, "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz")';
-        $xpath = sprintf('descendant-or-self::input[((contains(%s, "submit") or contains(%1$s, "button")) and contains(concat(\' \', normalize-space(string(@value)), \' \'), %s)) ', $translate, static::xpathLiteral(' '.$value.' ')).
-                         sprintf('or (contains(%s, "image") and contains(concat(\' \', normalize-space(string(@alt)), \' \'), %s)) or @id=%s or @name=%s] ', $translate, static::xpathLiteral(' '.$value.' '), static::xpathLiteral($value), static::xpathLiteral($value)).
-                         sprintf('| descendant-or-self::button[contains(concat(\' \', normalize-space(string(.)), \' \'), %s) or @id=%s or @name=%s]', static::xpathLiteral(' '.$value.' '), static::xpathLiteral($value), static::xpathLiteral($value));
-
-        return $this->filterRelativeXPath($xpath);
+        return $this->filterRelativeXPath(
+            sprintf(
```
Comment (javiereguiluz) [inline src/Symfony/Component/DomCrawler/Crawler.php]: Super minor comment: I think that our code policy requires that the `sprintf()` statement is in a single line, no matter how long it is (as done in #26485). Should we change this then?

Comment (Simperfit) [reply src/Symfony/Component/DomCrawler/Crawler.php]: I tend to agree to change this since it will be easier to read it.

**Review by Simperfit — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]
(I suppose :) )

**Comment by nicolas-grekas** [comment]
Thank you @dunglas.

**Pull request closed without merging**